src/table.rs

```diff
@@ -480,10 +554,129 @@ mod tests {
+        }
+        table.players[4].fold();
+        table.players[5].fold();
+        table
+    }
+
+    fn deal_test_cards_tied_best() -> Table {
+        let mut table = deal_test_cards();
+        if let PlayerState::Active(active) = &mut table.players[0].player_state {
+            active.hand = [poker::Card::new(poker::Rank::Ace, poker::Suit::Hearts), poker::Card::new(poker::Rank::King, poker::Suit::Hearts)];
+        }
+        if let PlayerState::Active(active) = &mut table.players[1].player_state {
+            active.hand = [poker::Card::new(poker::Rank::Ace, poker::Suit::Diamonds), poker::Card::new(poker::Rank::King, poker::Suit::Diamonds)];
+        }
+        table
+    }
+
+    fn deal_test_cards_tied_best_side_pot() -> Table {
+        let mut table = deal_test_cards_tied_best();
+        table.players[0].total_money = 0;
+        table.players[1].total_money = 0;
+        table
+    }
+
+    pub fn check_table_has_right_amount(table: &Table) {
+        let player_amount = table.players.iter().map(|x| x.total_money).sum::<i32>();
+        let pot_size = table.get_pot_size();
+        let table_sum = player_amount + pot_size;
+        assert_eq!(table_sum, table.players.len() as i32 * DEFAULT_START_MONEY);
+    }
+
+    #[test]
+    pub fn test_side_pot() {
+        let mut table = deal_test_cards_tied_best_side_pot();
+        table.current_player_index = 0;
+        table.dealer_button_index = table.players.len() - 1;
+        table.update_current_player_index_to_next_active();
+        assert_eq!(2, table.current_player_index);
+        table.take_action(HandAction::Raise(1));
+        assert_eq!(3, table.current_player_index);
+        table.take_action(HandAction::Call);
+        // assert_eq!(1, table.current_player_index);
+        // table.take_action(HandAction::Call);
+        // assert_eq!(2, table.current_player_index);
+        // table.take_action(HandAction::Call);
+        for i in 2..4 {
+            assert_eq!(table.table_state, Flop);
+            assert_eq!(table.current_player_index, i);
+            table.take_action(HandAction::Check);
+        }
+        for i in 2..4 {
+            assert_eq!(table.table_state, Turn);
+            assert_eq!(table.current_player_index, i);
+            table.take_action(HandAction::Check);
+        }
+        for i in 2..4 {
+            assert_eq!(table.table_state, River);
+            assert_eq!(table.current_player_index, i);
+            table.take_action(HandAction::Check);
+        }
+        assert_eq!(table.table_state, PreFlop);
+        assert!(table.players[0].is_alive());
+        assert!(table.players[1].is_alive());
+        assert_eq!(table.players[0].total_money, 2);
+        assert_eq!(table.players[1].total_money, 2);
+        assert_eq!(table.players[2].total_money, 499);
+        assert_eq!(table.players[3].total_money, 497);
+        assert_eq!(table.players[4].total_money, 498);
+        assert_eq!(table.players[5].total_money, 498);
+    }
```

Comment:
I think this looks good. I know it's a pain, but do we also want a test with multiple side pots? I figure if there are issues, they would come up there.

Reply:
OK FINE OMG ITS IN LETS JUST BE GOOD PROGRAMMERS OUT HERE SO HAPPY FOR US